src/rdf4cpp/rdf/IRI.cpp

```diff
@@ -100,13 +101,10 @@ IRI::operator datatypes::registry::DatatypeIDView() const noexcept {
     }
 }
 
-bool IRI::serialize(void *const buffer, writer::Cursor *cursor, writer::FlushFunc const flush) const noexcept {
+bool IRI::serialize(void *const buffer, writer::Cursor *cursor, writer::FlushFunc const flush, const writer::SerializationState* state) const noexcept {
```

Comment:
Just to reiterate what we discussed in the meeting and have it here: Currently we don't want to shorten arbitrary IRIs. So you can restore the interface to its previous version